libtsuba/include/katana/TxnContext.h

```diff
@@ -78,6 +86,12 @@ class KATANA_EXPORT TxnContext {
   bool GetTopologyWrite() const { return topology_write_; }
 
 private:
+  std::string ConcatRDGProperty(std::string rdg_dir, std::string prop) {
+    return rdg_dir + kPropSeparator + prop;
```

Comment:
I use `katana::Join` or `katana::JoinPath` in `URI.h`.

Reply:
This is the kind of operation that is unsafe to do on bare URIs.

Reply:
Well, then Yige, I guess my suggestion to use `JoinPath `is a requirement.

Reply:
@yige-hu can you please update this to use `Uri::Join` unless you know that `rdg_dir` did not come from `Uri::string`?

Reply:
@witchel I feel here '+' operator can work better because I'm concatenating three strings. `katana::JoinPath(rdg_dir, kPropSeparator).Join(prop)` doesn't look very straightforward.

Reply:
Wouldn't the call be `katana::JoinPath(rdg_dir, prop)`?

As Donald writes, if `kPropSeparator `or `prop` contained non-safe URI characters, the result would be an invalid URI.  You could argue that you are building a string, not a URI, but honestly, I think you are better off storing URIs.  I use them as cache keys.  Because they have a normalized form, they are better than a string, which will fail to match `gs://bucket/ `and `gs://bucket`, even though those two refer to the same storage location.